File: src/lang/SimpleToken.java
package lang;

public class SimpleToken extends Token {
    private TokenType type;                // 上のどのトークンか
    private String text;                // 切り出したトークンの綴り
    private int lineNo;                // このトークンがあった行
    private int colNo;                // このトークンがあった桁

    @Override
    public TokenType getType() {
        return type;
    }

    @Override
    public String getText() {
        return text == null ? "(null)" : text;
    }

    @Override
    public int getLineNo() {
        return lineNo;
    }

    @Override
    public int getColumnNo() {
        return colNo;
    }

    public int getIntValue() {
        return Integer.decode(text).intValue();
    }

    public SimpleToken(TokenType type, int lineNo, int colNo, String s) {
        this.type = type;
        this.lineNo = lineNo;
        this.colNo = colNo;
        this.text = s;
    }
}

File: src/lang/SimpleToken.kt
package lang

open class SimpleToken(// 上のどのトークンか
        override val type: TokenType, // このトークンがあった行
        override val lineNo: Int, // このトークンがあった桁
        override val columnNo: Int, // 切り出したトークンの綴り
        override val text: String) : Token() {

    val intValue: Int
        get() = Integer.decode(text).toInt()

}